rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp

```diff
@@ -70,6 +101,271 @@ TEST_F(RecordFixture, record_end_to_end_test) {
   EXPECT_THAT(wrong_topic_messages, IsEmpty());
 }
 
+// TODO(zmichaels11): Fix and enable this test on Windows.
+// This tests depends on the ability to read the metadata file.
+// Stopping the process on Windows does a hard kill and the metadata file is not written.
+#ifndef _WIN32
+TEST_F(RecordFixture, record_end_to_end_with_splitting_metadata_contains_all_topics) {
+  constexpr const char message_str[] = "Test";
+  constexpr const int bagfile_split_size = 4 * 1024 * 1024;  // 4MB.
+  constexpr const int message_size = 1024 * 1024;  // 1MB
+  constexpr const int expected_splits = 4;
+  constexpr const int message_count = bagfile_split_size * expected_splits / message_size;
+  constexpr const char first_topic_name[] = "/test_topic0";
+  constexpr const char second_topic_name[] = "/test_topic1";
+
+  std::stringstream command;
+  command << "ros2 bag record" <<
+    " --output " << root_bag_path_ <<
+    " --max-bag-size " << bagfile_split_size <<
+    " -a";
+  auto process_handle = start_execution(command.str());
+  wait_for_db();
+
+  {
+    const auto message = create_string_message(message_str, message_size);
+    const auto node = std::make_unique<rclcpp::Node>(
+      "TestMessagePublisher",
+      rclcpp::NodeOptions().start_parameter_event_publisher(false));
+    const auto first_publisher =
+      node->create_publisher<test_msgs::msg::Strings>(first_topic_name, 10);
+    rclcpp::WallRate message_rate{50ms};
+    constexpr const int message_batch_size = message_count / 2;
+
+    // Send first half of messages to /test_topic0
+    for (int i = 0; rclcpp::ok() && i < message_batch_size; ++i) {
```

Comment:
This looks like it can be made into a separate function.
Maybe a test fixture function.